source/game/kart/KartState.cc

```diff
@@ -201,16 +205,28 @@ void KartState::calcCollisions() {
         wallNrm += colData.noBounceWallNrm;
     }
 
+    m_wallBonkTimer = std::max(0, m_wallBonkTimer - 1);
+
+    if (colData.bWall) {
+        m_bWallCollision = true;
+
+        if (!wasWallCollision) {
+            m_bWallCollisionStart = true;
+        }
+
+        m_wallBonkTimer = 2;
+    }
+
     if (m_hwgTimer == 0 && colData.movement.y > 1.0f) {
         EGG::Vector3f movement = colData.movement;
         movement.normalise();
 
         if (movement.dot(EGG::Vector3f::ey) > 0.8f &&
                 colData.floorNrm.dot(EGG::Vector3f::ey) > 0.85f &&
-                movement.dot(colData.floorNrm) < 0.0f) {
-            colData.floorNrm.y = 0.0f;
-            colData.floorNrm.normalise();
-            wallNrm = colData.floorNrm;
+                movement.dot(colData.wallNrm) < 0.0f) {
```

Comment:
This is not supposed to be a true dot product, as both y values should be set to 0.0f.

Reply:
For now I'll expand it out, even though we can't find out what the compiler optimization is.